shop_basket/views.py

```diff
@@ -1,73 +1,84 @@
 from django.shortcuts import render, redirect
 from django.views.generic import TemplateView
 from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.contrib import messages
+from django.db.models import Max, Avg, Sum, Q, F, CharField
+from django.db.models.functions import Length
+
+from typing import List, Dict, Any
 
 from .models import Basket
 
+from .forms import BasketForm
+
 # Create your views here.
 
 
 class Index(TemplateView):
     template_name = 'index.html'
 
 
-@login_required(login_url='/accounts/login/')
-def basket_page(request):
-    return render(request, 'basket_page.html', {})
+class BasketPage(LoginRequiredMixin, TemplateView):
+    template_name = 'basket_page.html'
 
+    def get(self, request):
+        form = BasketForm()
+        return render(request, 'basket_page.html', {'form': form})
 
-def add_stuff_to_basket(request, pk):
-    price = request.POST['price']
-    title = request.POST['title']
 
-    if len(title) <= 1:
-        messages.error(request, 'title too short')
-        return redirect('/basket_page/')
+class BasketView(LoginRequiredMixin, TemplateView):
+    template_name = 'basket_page.html'
+    extra_context = {}
 
-    if price == '' or int(price) < 0 or int(price) == 0:
-        messages.error(request, 'price incorect')
-        return redirect('/basket_page/')
+    def get(self, request, *args, **kwargs):
+        self.extra_context['form'] = BasketForm()
 
-    if request.method == 'POST':
-        Basket.objects.create(user_id=pk, price=price, title=title).save()
+        return super(BasketView, self).get(request)
 
-        return redirect('/statistic_page/{}'.format(request.user.id))
-    else:
-        return render(request, 'basket_page.html', {})
+    def post(self, request, pk: int):
+        form = BasketForm(request.POST)
```

Comment:
Maybe it makes sense to do the following renaming of the class.
```suggestion
        form = OrderForm(request.POST)
```